File: Dummy/hawkshaw/library/datalayer/models/accessibility/NotificationKt.java
package com.hawkshaw.library.datalayer.models.accessibility;

import Y1.K;
import com.hawkshaw.library.datalayer.models.accessibility.PushNotificationRequest;
import com.hawkshaw.library.datalayer.room.notification.NotificationEntity;

public final class NotificationKt {
    public static final PushNotificationRequest.Notification toRequest(NotificationEntity notificationEntity) {
        K.n(notificationEntity, "<this>");
        return new PushNotificationRequest.Notification(notificationEntity.getPackageName(), notificationEntity.getExtraText(), notificationEntity.getExtraTitle(), notificationEntity.getExtraMessage(), notificationEntity.getExtraBigText(), notificationEntity.getExtraTextLines(), notificationEntity.getTickerText(), notificationEntity.getTimestamp());
    }
}

File: app/src/main/java/com/hawkshaw/library/datalayer/models/accessibility/NotificationKt.kt
package com.hawkshaw.library.datalayer.models.accessibility

import com.hawkshaw.library.datalayer.room.notification.NotificationEntity

/**
 * Extension function to convert a NotificationEntity to a PushNotificationRequest.Notification
 *
 * @return A new PushNotificationRequest.Notification instance with values from this NotificationEntity
 */
fun NotificationEntity.toRequest(): PushNotificationRequest.Notification {
    return PushNotificationRequest.Notification(
        id = this.id,
        packageName = this.packageName,
        title = this.title,
        text = this.text,
        timestamp = this.timestamp,
        pushed = this.pushed
    )
} 